pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -461,10 +462,11 @@ public void sendAsync(Message<?> message, SendCallback callback) {
                     sequenceId = msgMetadata.getSequenceId();
                 }
                 String uuid = totalChunks > 1 ? String.format("%s-%d", producerName, sequenceId) : null;
+                ChunkedMessageCtx chunkedMessageCtx = ChunkedMessageCtx.RECYCLER.get();
```

Comment:
It's better to avoid exposing the RECYCLER directly, the recycler should be an internal concept of the ChunkedMessageCtx.

Reply:
Only create the ctx if chunk message enabled?

Reply:
Thanks. Fixed